src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,172 @@
+import styled from '@emotion/styled'
+import scrollIntoView from 'scroll-into-view-if-needed'
+import React, { ReactNode, useState } from 'react'
+import { color, space, radius, device } from '../../theme'
+import { ChevronLeft, ChevronRight } from '../../icons'
+import { H2, H6 } from '../Heading'
+import { Text } from '../Text'
+
+const CarouselContainer = styled.div`
+  overflow: hidden;
+`
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin-block-end: ${space[8]};
+`
+const HeaderText = styled.div`
+  cursor: pointer;
+`
+const StyledH6 = styled(H6)`
+  color: ${color.action};
+  text-transform: none;
+  cursor: pointer;
+  padding-inline-start: ${space[8]};
+`
+const StyledText = styled(Text)`
+  color: ${color.foregroundMuted};
+`
+const Controls = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${space[8]};
+`
+const Control = styled.button`
+  display: none;
+
+  @media ${device.tablet} {
+    inline-size: ${space[32]};
+    block-size: ${space[32]};
+    border-radius: ${radius.md};
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &:hover {
+      background-color: ${color.actionBackground};
+    }
+
+    &:disabled {
+      opacity: 0.4;
+      cursor: default;
+    }
+  }
+`
+interface CardProps {
+  amountOfCards: number
+  numberOfColums: number
+}
+
+const Cards = styled.div<CardProps>`
+  display: grid;
+  grid-template-columns: ${({ amountOfCards }) =>
+    `repeat(${amountOfCards}, 1fr)`};
+  width: 100%;
+  overflow-x: scroll;
+  overflow-y: hidden;
+  scrollbar-width: none;
+  grid-gap: ${space[16]};
+  scroll-behavior: smooth;
+  scroll-snap-type: x mandatory;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+
+  @media ${device.tablet} {
+    overflow-x: inherit;
+  }
+`
+
+interface Props {
+  title: string
+  id: string
+  subtitle?: string
+  viewAll?: string
```

Comment:
`viewAll` is a bit vague for what it represents I think. Maybe something like `viewAllLabel`?